src/WaterPokemon.java

```diff
@@ -0,0 +1,38 @@
+package src;
+
+public abstract class WaterPokemon extends Pokemon {
+    private String type;
+    public WaterPokemon(double healthPoint, double experiencePoints) {
+        super(healthPoint, experiencePoints);
+    }
+
+    public WaterPokemon(double healthPoint, double experiencePoints, String name
```

Comment:
Je constructor is wat lang en daardoor wat moeilijk leesbaar. In dat soort gevallen is het misschien handig om sommige wardes te gaan groeperen in hun eigen klasse. Dus ipv "WaterPokemon(double healthPoint, double experiencePoints, String name, int level, double attack, double defence, double accuracy, double weight, double height)" krijg je dan misschien iets als "WaterPokemon(double healthPoint, double experiencePoints, String name, PokemonStats stats). Hoewel.. soms is het ook niet te voorkomen dat je lange constructors krijgt